ui/app/AppMain.qml

```diff
@@ -800,26 +802,22 @@ Item {
             width: 350
             x: parent.width / 2 - width / 2
             y: parent.height / 2 - height / 2
-            // TODO improve this to work with community Chats as well
-            modelList: mainModule.getChatSectionModule().model
-            getText: function (modelData) {
-                return modelData.name
-            }
-            getId: function (modelData) {
-                return modelData.itemId
-            }
-            getImageComponent: function (parent, modelData) {
-                return statusSmartIdenticonComponent.createObject(parent, {
-                                                                     imageSource: modelData.identicon,
-                                                                     name: modelData.name
-                                                            });
-            }
-
-            onClicked: function (index, id) {
-                Global.changeAppSectionBySectionType(Constants.appSection.chat)
-                mainModule.getChatSectionModule().setActiveItem(id, "")
+            modelList: rootStore.mainModuleInst.getAllChatsModel()
```

Comment:
If possible, you can expose the `allChatsModel` from the store directly, instead of passing by `mainModuleInst`. So something like this:
```qml
 modelList: rootStore.allChatsModel 
```

And in the store you have
```qml
allChatsModel : mainModuleInst.allChatsModel // This will only work when the QtProperty will be exposed
```